worker/worker.go

```diff
@@ -912,19 +907,17 @@ func (w *worker) objectsKeyHandlerPUT(jc jape.Context) {
 		var slowHosts []int
 
 		lr := io.LimitReader(cr, int64(rs.MinShards)*rhpv2.SectorSize)
-		if err := w.withHosts(ctx, contracts, func(hosts []sectorStore) (err error) {
-			// move slow hosts to the back of the array
-			sort.SliceStable(hosts, func(i, j int) bool {
-				return slow[hosts[i].PublicKey()] < slow[hosts[j].PublicKey()]
-			})
-
-			// upload the slab
-			s, length, slowHosts, err = uploadSlab(ctx, lr, uint8(rs.MinShards), uint8(rs.TotalShards), hosts, &tracedContractLocker{w.bus}, w.uploadSectorTimeout)
-			for _, h := range slowHosts {
-				slow[hosts[h].PublicKey()]++
-			}
-			return err
-		}); err == io.EOF {
+		// move slow hosts to the back of the array
+		sort.SliceStable(contracts, func(i, j int) bool {
+			return slow[contracts[i].HostKey] < slow[contracts[j].HostKey]
+		})
+
+		// upload the slab
+		s, length, slowHosts, err = uploadSlab(ctx, w, lr, uint8(rs.MinShards), uint8(rs.TotalShards), contracts, &tracedContractLocker{w.bus}, w.uploadSectorTimeout)
```

Comment:
`nit` definitely not important to this PR but Luke shared an [article](https://lexi-lambda.github.io/blog/2019/11/05/parse-don-t-validate/) yesterday and this is _exactly_ what it said _not_ to do 😛